test/e2e/multikueue/e2e_test.go

```diff
@@ -372,6 +372,93 @@ var _ = ginkgo.Describe("MultiKueue", func() {
 				}
 			})
 		})
+		ginkgo.It("Should create a pod group on worker if admitted", func() {
+			numPods := 2
+			groupName := "test-group"
+			group := testingpod.MakePod(groupName, managerNs.Name).
+				Queue(managerLq.Name).
+				Image(util.E2eTestAgnHostImage, util.BehaviorExitFast).
+				RequestAndLimit(corev1.ResourceCPU, "200m").
+				RequestAndLimit(corev1.ResourceMemory, "1G").
+				MakeGroup(numPods)
+
+			ginkgo.By("Creating the Pod group", func() {
+				for _, p := range group {
+					gomega.Expect(k8sManagerClient.Create(ctx, p)).To(gomega.Succeed())
+					gomega.Expect(p.Spec.SchedulingGates).To(gomega.ContainElements(
+						corev1.PodSchedulingGate{Name: podconstants.SchedulingGateName},
+					))
+				}
+			})
+
+			// Since it requires 2G of memory, this pod group can only be admitted in worker 2.
+			pods := &corev1.PodList{}
+			ginkgo.By("ensure all pods are created", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sManagerClient.List(ctx, pods, client.InNamespace(managerNs.Name))).To(gomega.Succeed())
+					g.Expect(pods.Items).Should(gomega.HaveLen(numPods))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			createdLeaderWorkload := &kueue.Workload{}
+			wlLookupKey := types.NamespacedName{Name: groupName, Namespace: managerNs.Name}
+
+			// the execution should be given to worker2
+			ginkgo.By("Waiting to be admitted in worker2", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sManagerClient.Get(ctx, wlLookupKey, createdLeaderWorkload)).To(gomega.Succeed())
+					g.Expect(workload.FindAdmissionCheck(createdLeaderWorkload.Status.AdmissionChecks, multiKueueAc.Name)).To(gomega.BeComparableTo(&kueue.AdmissionCheckState{
+						Name:    multiKueueAc.Name,
+						State:   kueue.CheckStatePending,
+						Message: `The workload got reservation on "worker2"`,
+					}, cmpopts.IgnoreFields(kueue.AdmissionCheckState{}, "LastTransitionTime")))
+				}, util.LongTimeout, util.Interval).Should(gomega.Succeed())
+
+				ginkgo.By("ensure all pods are created", func() {
+					gomega.Eventually(func(g gomega.Gomega) {
+						g.Expect(k8sManagerClient.List(ctx, pods, client.InNamespace(managerNs.Name))).To(gomega.Succeed())
+						for _, p := range pods.Items {
+							g.Expect(utilpod.HasGate(&p, podconstants.SchedulingGateName)).To(gomega.BeTrue())
+						}
+					}, util.Timeout, util.Interval).Should(gomega.Succeed())
+				})
+			})
+			finishReason := "PodCompleted"
+			finishPodConditions := []corev1.PodCondition{
```

Comment:
I think asserting on the exact list of conditions is not a good practice, because this can change between k8s versions - yes it is fairly stable, but can cause unnecessary failures on k8s upgrades.

I think just checking the PodPhase is enough, as it is part of status and the status is updated atomically with conditions.